src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,40 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.util.Storage;
+import duke.util.TaskList;
+import duke.util.Ui;
+
+/**
+ * Represents a command in the Duke program.
+ */
+public abstract class Command {
+    /**
+     * Defines the execution of this command.
+     *
+     * @param tasks   Tasks of the Duke program.
+     * @param ui      Ui of the Duke program.
+     * @param storage Storage of the Duke program.
+     * @throws DukeException If there is/are invalid argument(s) passed into this command,
+     *                       or changes cannot be saved to storage.
+     */
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException;
+
+    /**
+     * Returns the response of this command.
+     *
+     * @param tasks   Tasks of the Duke program.
+     * @param ui      Ui of the Duke program.
+     * @param storage Storage of the Duke program.
+     * @throws DukeException If there is/are invalid argument(s) passed into this command,
+     *                       or changes cannot be saved to storage.
+     */
+    public abstract String executeAndGetResponse(TaskList tasks, Ui ui, Storage storage) throws DukeException;
```

Comment:
This function seems to be essentially equivalent to `execute` other than the fact that it has a `String` return type. Was there any specific reason why you chose to have a separate abstract function, rather than to change the return type of `execute` and return empty/null strings if there is no response?

Looking at the classes that extend from `Command`, it seems that this has led to multiple [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) violations. I think reworking your codebase to have just the `execute` method would be beneficial for code quality.

Reply:
This was a quick fix after adding the GUI as the response of each command is to be returned, unlike in the CLI, the response is just to be printed on the console. However, your comment gave me an idea to refactor my code. Something as follows:
````java
public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
    String response = executeAndGetResponse(tasks, ui, storage);
    ui.showResponse(response);
}
````
Thanks!